### PR TITLE
fix: consistent link preview component

### PR DESCRIPTION
**Pull request by Enigma-I-am**
## 🎯 Goal
This PR fixes the inconsistent styles of the link message components and implement the design guide.

## 🛠 Implementation details
I this PR I styled of the card component and to be more consistent with the [figma](https://www.figma.com/file/oeHiSMDoutuVbJ9WiVyhcG/Chat-Design-Kit-1.5-Android---In-Progress?node-id=17497%3A140906) and indication youtube video links are video by adding a play button stack on top of the card

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [ ] New code is covered by tests
- [x] Screenshots added for visual changes
- [ ] Documentation is updated

### CODE REVIEW

File: package/src/components/Attachment/Card.tsx [hunk 1]
```diff
@@ -62,6 +65,15 @@ const styles = StyleSheet.create({
     fontSize: 12,
     marginHorizontal: 8,
   },
+  roundedView: {
```
Comment (Enigma-I-am) [inline package/src/components/Attachment/Card.tsx:68]: @madsroskar

File: package/src/components/Attachment/Card.tsx [hunk 2]
```diff
@@ -41,13 +42,15 @@ const styles = StyleSheet.create({
   },
   authorNameMask: {
     bottom: 0,
-    left: 8,
+    left: 2,
```
Comment (Enigma-I-am) [inline package/src/components/Attachment/Card.tsx:45]: @madsroskar

File: package/src/components/Attachment/Card.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ import {
   useMessagesContext,
 } from '../../contexts/messagesContext/MessagesContext';
 import { useTheme } from '../../contexts/themeContext/ThemeContext';
+import { Play } from '../../icons/Play';
```
Comment (Enigma-I-am) [inline package/src/components/Attachment/Card.tsx:27]: @madsroskar

File: package/src/components/Attachment/Card.tsx [hunk 4]
```diff
@@ -116,11 +128,15 @@ const CardWithContext = <
     text,
     thumb_url,
     title,
+    type,
```
Comment (Enigma-I-am) [inline package/src/components/Attachment/Card.tsx:131]: @madsroskar

File: package/src/components/Attachment/Card.tsx [hunk 5]
```diff
@@ -143,6 +159,8 @@ const CardWithContext = <
 
   const defaultOnPress = () => !error && openURL && openURL();
 
+  const isVideo = type === 'video';
+
```
Comment (Enigma-I-am) [inline package/src/components/Attachment/Card.tsx:163]: @madsroskar

File: package/src/components/Attachment/Card.tsx [hunk 6]
```diff
@@ -180,11 +198,21 @@ const CardWithContext = <
       {CardCover && <CardCover {...props} />}
       {uri && !CardCover && (
         <View>
-          <Image
+          <ImageBackground
```
Comment (Enigma-I-am) [inline package/src/components/Attachment/Card.tsx:201]: @madsroskar

File: package/src/components/Attachment/Card.tsx [hunk 7]
```diff
@@ -180,11 +198,21 @@ const CardWithContext = <
...
+            imageStyle={styles.cardCover}
             resizeMode='cover'
             source={{ uri: makeImageCompatibleUrl(uri) }}
             style={[styles.cardCover, cover, stylesProp.cardCover]}
-          />
+          >
```
Comment (Enigma-I-am) [inline package/src/components/Attachment/Card.tsx:206]: @madsroskar

**Review by madsroskar — CHANGES_REQUESTED** [review]

**Review by madsroskar — CHANGES_REQUESTED** [review]

File: package/src/contexts/themeContext/utils/theme.ts [hunk 8]
```diff
@@ -196,7 +196,7 @@ export type Theme = {
       rightContainer: ViewStyle;
       usernameText: TextStyle;
     };
-    videoControl: {
```
Comment (vanGalilea) [inline package/src/contexts/themeContext/utils/theme.ts:199]: This is a breaking change guys 🤷🏻 What are we planning here?

**Comment by vishalnarkhede** [comment]
Whats the reason for this refactor btw? The refactor seems to be on top of what Khushal just built and merged. 

I think we should close this PR

**Comment by Enigma-I-am** [comment]
@vishalnarkhede this PR only handled the the styling of the link preview components. My changes where only made in only the made in the card.tsx component

**Comment by vanGalilea** [comment]
We still have inconsistencies see screenshots below:
Some of the links are with a name tab (circled in red) and some aren't, is this intentional? @Enigma-I-am can you compare it to Fra's design?

<table>
    <thead>
        <tr>
            <td>Youtube/GH links with name label</td>
            <td>getstream and NOS links without name label</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>

            </td>
            <td>

            </td>
        </tr>
    </tbody>
</table>

**Review by vanGalilea — CHANGES_REQUESTED** [review]

**Comment by Enigma-I-am** [comment]
> We still have inconsistencies see screenshots below: Some of the links are with a name tab (circled in red) and some aren't, is this intentional? @Enigma-I-am can you compare it to Fra's design?
> 
> Youtube/GH links with name label	getstream and NOS links without name label
> 	

@vanGalilea for that component we check if the author name exists and if so we render. It is returned from the client and some companies don't provide the author name just like how some don't provide an image

File: package/src/components/Attachment/Card.tsx [hunk 9]
```diff
@@ -133,6 +149,9 @@ const CardWithContext = <
           footer: { description, title: titleStyle, ...footerStyle },
           noURI,
         },
+        content: {
+          playIcon: { height, pathFill, width },
```
Comment (vishalnarkhede) [inline package/src/components/Attachment/Card.tsx:153]: This should be nested under `card`

File: package/src/components/Attachment/Card.tsx [hunk 10]
```diff
@@ -62,6 +65,15 @@ const styles = StyleSheet.create({
     fontSize: 12,
     marginHorizontal: 8,
   },
+  playButtonStyle: {
+    alignItems: 'center',
+    backgroundColor: '#fff',
```
Comment (vishalnarkhede) [inline package/src/components/Attachment/Card.tsx:70]: Use this from theme color pallete

File: package/src/contexts/themeContext/utils/theme.ts [hunk 11]
```diff
@@ -909,6 +910,11 @@ export const defaultTheme: Theme = {
       metaText: {
         fontSize: 12,
       },
+      playIcon: {
+        height: 24,
+        pathFill: '#000',
```
Comment (vishalnarkhede) [inline package/src/contexts/themeContext/utils/theme.ts:915]: Use this from theme color pallete

**Comment by khushal87** [comment]
@Enigma-I-am Can you explain in the PR what are the changes are introduced in this PR? Will be happy to test them while reviewing.

File: package/src/components/Attachment/Card.tsx [hunk 12]
```diff
@@ -116,11 +128,15 @@ const CardWithContext = <
     text,
     thumb_url,
     title,
+    type,
   } = props;
 
   const {
     theme: {
       colors: { accent_blue, black, blue_alice, transparent },
+      imageGallery: {
```
Comment (khushal87) [inline package/src/components/Attachment/Card.tsx:137]: This should come from Card. And the name can be `playButtonStyle` as Vish mentioned.

File: package/src/contexts/themeContext/utils/theme.ts (hunk 11 above)
Comment (khushal87) [reply package/src/contexts/themeContext/utils/theme.ts:915]: I might be wrong but I prefer having the default styles in the component itself. Though we can keep this here but ideally keeping the default styles in the component makes more sense.

**Review by khushal87 — CHANGES_REQUESTED** [review]
Note: The link is out of the view in Reply

**Comment by vishalnarkhede** [comment]
> 1. The link is out of Reply view.
> 2. The thumbnail image is displayed and the video thumbnail with a play icon is also displayed. I think there should only be one at a time for the case of links.
> 
> 

This issue is also present on develop branch.

File: package/src/components/Attachment/Card.tsx [hunk 13]
```diff
@@ -180,11 +201,18 @@ const CardWithContext = <
...
         <View>
-          <Image
+          <ImageBackground
+            imageStyle={styles.cardCover}
             resizeMode='cover'
             source={{ uri: makeImageCompatibleUrl(uri) }}
             style={[styles.cardCover, cover, stylesProp.cardCover]}
-          />
+          >
+            {isVideo ? (
```
Comment (khushal87) [inline package/src/components/Attachment/Card.tsx:210]: `isVideo && !lastAttachment.og_scrape_url` will solve your problem. I think this came from the video PR. Sorry for that!

Comment (khushal87) [reply package/src/components/Attachment/Card.tsx:210]: Fixed this - #1434

Comment (Enigma-I-am) [reply package/src/components/Attachment/Card.tsx:210]: I am merging the develop into this

Comment (vishalnarkhede) [reply package/src/components/Attachment/Card.tsx:208]: You need to make similar change here @Enigma-I-am . The PR that @khushal87 shared only updates `Reply` component

Comment (Enigma-I-am) [reply package/src/components/Attachment/Card.tsx:208]: This change made the play button disappear totally. went with `isVideo && og_scrape_url` instead

**Review by vishalnarkhede — APPROVED** [review]

File: package/src/components/Attachment/Card.tsx (hunk 13 above)
Comment (Enigma-I-am) [reply package/src/components/Attachment/Card.tsx]: Ohhh... I understand now you meant `isVideo && lastAttachment.og_scrape_url` and not `isVideo && !lastAttachment.og_scrape_url`

**Comment by vishalnarkhede** [comment]
Lets merge it after the current release :) cc @vanGalilea

**Pull request merged**